src/services/api/data.js

```diff
@@ -0,0 +1,43 @@
+import defaultClient from '@/services/api/defaultClient.js';
+import endpoint from '@/services/api/endpoint.js';
+
+export async function getAllMoviesData() {
+  const moviesResponse = await defaultClient.get(endpoint.MOVIES);
+  return moviesResponse.data;
+}
+
+export async function getMovieData(movieId) {
+  const movieResponse = await defaultClient.get(`${endpoint.MOVIES}/${movieId}`);
+  return movieResponse.data;
+}
+
+export async function getMovieGenresData() {
+  const genreResponse = await defaultClient.get(`${endpoint.GENRES}`);
+  return genreResponse.data;
+}
```

Comment:
Same here 😄

Reply:
In this scenario actually it causes a lot of errors:
```
export async function getAllMoviesData() {
  return await defaultClient.get(endpoint.MOVIES)?.data;
}
```
But something like this works fine if you would like it more:
```
export async function getAllMoviesData() {
  const { data } = await defaultClient.get(endpoint.MOVIES);
  return data;
}
```

And most of my logic in many other places is based on that those functions return the data property, hence the naming convention "getSomethingData"

Reply:
Ah right you return the data here!
The destructuring approach is absolutely fine then, my mistake, I didn't catch that 😅

Reply:
I can also go with something like this since the code would repeat
```
async function getData(endpoint) {
  const { data } = await defaultClient.get(endpoint);
  return data;
}

export async function getAllMoviesData() {
  return getData(endpoint.MOVIES);
}

export async function getMovieData(movieId) {
  return getData(`${endpoint.MOVIES}/${movieId}`);
}

// ...
```

Reply:
Nah, there is no need unless you really want to - this is a bit overkill 😄

Reply:
The whole file would be as short as this right now:

```
import defaultClient from '@/services/api/defaultClient.js';
import endpoint from '@/services/api/endpoint.js';

const getData = async (endpoint, params = {}) => {
  const { data } = await defaultClient.get(endpoint, params);
  return data;
};

export const getAllMoviesData = () => getData(endpoint.MOVIES);

export const getMovieData = movieId => getData(`${endpoint.MOVIES}/${movieId}`);

export const getMovieGenresData = () => getData(endpoint.GENRES);

export const getUserData = () => getData(endpoint.USER);

export const getSeancesData = (movieId = null, date = null) =>
  getData(endpoint.SEANCES, {
    params: {
      movie_id: movieId,
      date: date,
    },
  });
```

Reply:
Sorry I did not see any of your further comments, page did not update for me in real time 😅

Reply:
And yes, overkill might be, yet I did it and I like it so let's leave it like that anyway 😅 Sorry again for missing your answers!